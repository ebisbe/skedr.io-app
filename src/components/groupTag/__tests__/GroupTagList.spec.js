import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Comp from '../GroupTagList.vue'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('GroupTagList.vue', () => {
  it('shows a group', () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      vuetify: new Vuetify(),
      store: new Vuex.Store({
        modules: {
          tagsFilter: {
            namespaced: true,
            state: {
              items: { tag1: true }
            }
          }
        }
      }),
      propsData: {
        group: {
          id: 'groupId',
          title: 'Group Title',
          poolCount: 1000000000,
          members: 200,
          icon: '',
          photoLimitOptOut: true
        },
        groupTag: {
          tags: ['tag1', 'tag2', 'tag3']
        }
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('tags watch must change manageTags to false', async () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      vuetify: new Vuetify(),
      store: new Vuex.Store({
        modules: {
          tagsFilter: {
            namespaced: true,
            state: {
              items: { tag1: true }
            }
          }
        }
      }),

      propsData: {
        group: {
          id: 'groupId',
          title: 'Group Title',
          photoLimitOptOut: true,
          icon: ''
        },
        groupTag: {
          tags: []
        }
      }
    })
    wrapper.vm.manageTags = true
    expect(wrapper.vm.manageTags).toBe(true)
    await wrapper.setProps({ groupTag: { tags: ['tag1'] } })
    expect(wrapper.vm.manageTags).toBe(false)
  })

  it('hovers over the element to show the link', done => {
    const wrapper = mount(Comp, {
      localVue,
      vuetify: new Vuetify(),
      propsData: {
        group: {
          id: 'groupId',
          title: 'Group Title',
          photoLimitOptOut: true,
          icon: ''
        },
        groupTag: {
          tags: []
        }
      },
      // Mocks is needed due to use mount() https://github.com/kazupon/vue-i18n/issues/323
      // otherwise should be handled by i18n_test.js configs
      mocks: {
        $i18n: {
          t: key => key
        },
        $t: key => key
      }
    })
    expect(wrapper.vm.hover).toBe(false)
    wrapper.find('div > div').trigger('mouseover')
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.hover).toBe(true)
      done()
    })
  })
})
