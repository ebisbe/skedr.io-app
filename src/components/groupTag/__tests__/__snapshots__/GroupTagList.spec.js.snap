// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupTagList.vue shows a group 1`] = `
<div>
  <!---->
  <v-list-item-stub activeclass="" link="true" tag="div">
    <v-list-item-avatar-stub size="40">
      <external-link-badge-stub href="https://www.flickr.com/groups/groupId"><img src=""></external-link-badge-stub>
    </v-list-item-avatar-stub>
    <v-list-item-content-stub>
      <v-list-item-title-stub>
        <photo-limit-opt-out-message-stub optout="true"></photo-limit-opt-out-message-stub> <strong>Group Title</strong>
      </v-list-item-title-stub>
      <v-list-item-subtitle-stub>
        <v-layout-stub tag="div" wrap="">
          <v-flex-stub tag="div" xs4="" sm2="" class="text-left">
            <v-icon-stub>photo</v-icon-stub>
            1000M
          </v-flex-stub>
          <v-flex-stub tag="div" xs4="" sm2="" class="text-left">
            <v-icon-stub>group</v-icon-stub>
            200
          </v-flex-stub>
          <v-flex-stub tag="div" xs4="" sm2="" class="text-left pt-1"></v-flex-stub>
          <v-flex-stub tag="div" xs12="" sm6="" class="text-right">
            <v-chip-stub small="true" activeclass="" tag="span" inputvalue="true" active="true" closeicon="$delete" filtericon="$complete" label="true" class="mx-1">
              <v-icon-stub color="accent" small="" left="" class="mr-1">label</v-icon-stub>
              tag1
            </v-chip-stub>
            <v-chip-stub small="true" activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" label="true" class="mx-1">
              <v-icon-stub small="" left="" class="mr-1">label</v-icon-stub>
              tag2
            </v-chip-stub>
            <v-chip-stub small="true" activeclass="" tag="span" active="true" closeicon="$delete" filtericon="$complete" label="true" class="mx-1">
              <v-icon-stub small="" left="" class="mr-1">label</v-icon-stub>
              tag3
            </v-chip-stub>
          </v-flex-stub>
        </v-layout-stub>
      </v-list-item-subtitle-stub>
    </v-list-item-content-stub>
    <v-list-item-action-stub>
      <v-tooltip-stub opendelay="0" closedelay="0" contentclass="" fixed="true" left="true" openonhover="true" maxwidth="auto" nudgebottom="0" nudgeleft="0" nudgeright="0" nudgetop="0" nudgewidth="0" tag="span"> <span></span></v-tooltip-stub>
    </v-list-item-action-stub>
    <group-tag-dialog-stub groupid="groupId" title="Group Title" tags="tag1,tag2,tag3"></group-tag-dialog-stub>
  </v-list-item-stub>
</div>
`;
